### PR TITLE
Fix for test failure (test_iodaconv_ghcn_snod) with spack-stack build

### PR DESCRIPTION
**Pull request by srherbener**
## Description

This PR fixes an issue that surfaced with the `test_iodaconv_ghcn_snod` test when using a spack-stack build. spack-stack builds the pandas python package using the python installation that is found when spack-stack install is running. The `test_iodaconv_ghcn_snod` test runs a python script `ghch_snod2ioda.py` which uses the pandas read_csv command to read a text file containing station names. The read_csv is called with '\n' as a delimiter, but this was failing on my Mac (and possibly other platforms). This issue is that read_csv with it's built-in parser does not recognize '\n' as a delimiter, so it switches to the 'python' parser which can vary depending on the version that is found when spack-stack runs.

I'm not expert with pandas, so there might be a much better way to solve this. Please treat this PR as a proposal, and feel free to suggest a better way to fix this. I changed the delimiter from '\n' to '\t' (tab) which is recognized by the read_csv built-in parser and appears to work (since the intent appears to be to get read_csv to treat the entire line as a single entry in the output dataframe.

### Issue(s) addressed

None

## Acceptance Criteria (Definition of Done)

The `test_iodaconv_ghcn_snod` test passes on the Mac (and all platforms).

## Dependencies

None

## Impact

None

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
This is odd... `\n` is newline, no? and `\t` is tab? Those aren't really interchangeable are they? I'll let @YoulongXia-NOAA comment as he presumably wrote this, I don't know the specifics of the input file. I can't imagine why `\n` would ever be used as a delimiter, because that would just be 'read each line'.

**Comment by YoulongXia-NOAA** [comment]
@tsga, I am taking AL for today and tomorrow. I used your python code to work for ioda-converter. There is one sentence using \n and Stephen wants to change \t, and Cory thought either is correct. You have more experience about GHCN snowD csv file. Could you please see how to fix this issue and give some explanations first?
 
Click File changed will see the modification from Stephen. I will take a look after I am back from AL.

**Comment by YoulongXia-NOAA** [comment]
@tsga , either is NOT correct.

**Comment by srherbener** [comment]
> This is odd... `\n` is newline, no? and `\t` is tab? Those aren't really interchangeable are they? I'll let @YoulongXia-NOAA comment as he presumably wrote this, I don't know the specifics of the input file. I can't imagine why `\n` would ever be used as a delimiter, because that would just be 'read each line'.

@CoryMartin-NOAA it looks like the code where the change is made is intending to build a pandas dataframe out of a text file (not a csv file). The text file has a different number of fields as you go line-by-line so the code (I believe) is reading the entire line as one field, and then in subsequent commands splits it up in an even (same number of fields per row in the data frame) manner. newline and tab delimiters of course are not the same, however the input file does not contain tabs so in this case they have the same result (ie, the entire line is treated as one field). Using a tab instead of a newline avoids the error I saw on my MacBook with my python3 (3.9.13) installation (when using spack-stack to build pandas).

I admit that this is a hack fix, so if someone more familiar with pandas has a better way to address this I am all for that. I think the idea is to transfer a text file that contains space separated fields, where there can be a different number of fields line-by-line, into a pandas dataframe.

**Comment by tsga** [comment]
@srherbener, yes that was the reason, but I only tested on Windows. 

For now (until/unless someone comes up with a better way to address it in Pandas) a quick fix would be to use  sep='\r\n'  which I think should work on Mac as well as on Windows (tested on Windows). The downside will be it will be slower and we'll get a parser warning about using regex as delimiter.

**Review by tsga — CHANGES_REQUESTED** [review]
Change line 105 to:
df10all = pd.read_csv(self.fixfile, header=None, sep='\r\n')

**Comment by srherbener** [comment]
> Change line 105 to: df10all = pd.read_csv(self.fixfile, header=None, sep='\r\n')

@tsga thank you for your review! Your suggestion works on my Mac, so I applied it.

**Review by CoryMartin-NOAA — APPROVED** [review]
Yes this should handle both 'return' cases on all platforms (I think)

**Comment by YoulongXia-NOAA** [comment]
I tested it on Hera and it works.

File: src/land/ghcn_snod2ioda.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def assignValue(colrowValue, df400):
         df30 = df30[df30["DATETIME"] == parse(startdate).date()]
         # Read station files
         cols = ["ID", "LATITUDE", "LONGITUDE", "ELEVATION", "STATE", "NAME", "GSN_FLAG", "HCNCRN_FLAG", "WMO_ID"]
-        df10all = pd.read_csv(self.fixfile, header=None, sep='\n')
+        df10all = pd.read_csv(self.fixfile, header=None, sep='\r\n')
```
Comment (YoulongXia-NOAA) [inline src/land/ghcn_snod2ioda.py:105]: Look good for me. I tested on Hera for ioda-bundle, ctest passed.

**Review by YoulongXia-NOAA — APPROVED** [review]

**Comment by YoulongXia-NOAA** [comment]
By the way, thank you very much @tsga.

**Review by tsga — APPROVED** [review]

**Pull request merged**